### PR TITLE
[W5][W11-3]Arielyte Tsen Chung Ming

### PR DESCRIPTION
**Pull request by chung-ming**
Ready for review.

### CODE REVIEW

File: src/seedu/addressbook/commands/SortCommand.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * Sorts all persons in the address book to the user.
+ */
+public class SortCommand extends Command {
+
+    public static final String  COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE= COMMAND_WORD
+            + ": Sorts all persons in the address book as a list with index numbers.\n"
+            + "Example: " + COMMAND_WORD;
+
```
Comment (nivenang) [inline src/seedu/addressbook/commands/SortCommand.java:17]: Double empty lines can be removed, same for line 6

Comment (chung-ming) [reply src/seedu/addressbook/commands/SortCommand.java:17]: Changes made to remove double empty lines

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 2]
```diff
@@ -81,6 +82,26 @@ public UniquePersonList(UniquePersonList source) {
...
 
+    /**
+     * Sorts a shadow list of persons by name in ascending order.
+     * Note that the original list is not modified.
+     *
+     * @return an unmodifiable List view with elements sorted and cast as immutable
+     */
+    public List<ReadOnlyPerson> sort() {
+        List<Person> shadowList = internalList;
+        shadowList.sort(new Comparator<Person>() {
```
Comment (kylase-learning) [inline src/seedu/addressbook/data/person/UniquePersonList.java:93]: How about implementing such sorting in OOP fashion? This method is more like FP.

File: test/java/seedu/addressbook/storage/StorageFileTest.java [hunk 3]
```diff
@@ -25,7 +25,8 @@
 import static seedu.addressbook.util.TestUtil.assertFileDoesNotExist;
 
 public class StorageFileTest {
-    private static final String TEST_DATA_FOLDER = "test/data/StorageFileTest";
+    private static final String TEST_DATA_FOLDER = "C:/Users/chungming/Google Drive/" +
```
Comment (kylase-learning) [inline test/java/seedu/addressbook/storage/StorageFileTest.java:28]: Hard-coding the path is not encouraged as other developers that work with you will not be able to run the code.

**Review by kylase-learning — COMMENTED** [review]
Great to see you have submitted your changes via your own branch and done the documentation. 👍

What if the user wants to sort in the reverse order?

You have demonstrated the progress of this exercise through the commits you have done. Maybe, as you get more comfortable, your commits can be done in batch, by committing the test, the implementation and the relevant documentation together, instead of committing changes in each file separately. This way, it is easier to navigate through commits in events you need to revert to certain state of the code, and all its related dependencies (such as documentation, tests) are in sync. Otherwise, for example, you can revert the commit for the code change but not the documentation (if the documentation is written prior to the code).

Please close this PR after reading comments. Thanks.

**Pull request closed without merging**